alcs-api/src/application/application.controller.ts

```diff
@@ -1,48 +1,87 @@
-import { Body, Controller, Delete, Get, Post } from '@nestjs/common';
+import { Body, Controller, Delete, Get, Patch, Post } from '@nestjs/common';
 import { ApiOAuth2 } from '@nestjs/swagger';
-import { ApplicationCreateDto, ApplicationDto } from './application.dto';
+import { ApplicationStatusService } from './application-status/application-status.service';
+import { ApplicationDto, ApplicationPartialDto } from './application.dto';
+import { Application } from './application.entity';
 import { ApplicationService } from './application.service';
 import * as config from 'config';
 
 @ApiOAuth2(config.get<string[]>('KEYCLOAK.SCOPES'))
 @Controller('application')
 export class ApplicationController {
-  constructor(private readonly applicationService: ApplicationService) {}
+  constructor(
+    private readonly applicationService: ApplicationService,
+    private applicationStatusService: ApplicationStatusService,
+  ) {}
 
   @Get()
   async getAll(): Promise<ApplicationDto[]> {
     const applications = await this.applicationService.getAll();
     return applications.map<ApplicationDto>((app) => {
       return {
-        number: app.number,
+        fileNumber: app.fileNumber,
         title: app.title,
         body: app.body,
-        status: {
-          code: app.status?.code,
-          description: app.status?.description,
-        },
+        status: app.status.code,
       };
     });
   }
 
   @Post()
-  async add(
-    @Body() application: ApplicationCreateDto,
+  async add(@Body() application: ApplicationDto): Promise<ApplicationDto> {
+    const entity = await this.mapToEntity(application);
+    const app = await this.applicationService.create(entity);
+    return {
+      fileNumber: app.fileNumber,
+      title: app.title,
+      body: app.body,
+      status: app.status.code,
+    };
+  }
+
+  @Patch()
+  async update(
+    @Body() application: ApplicationPartialDto,
   ): Promise<ApplicationDto> {
-    const app = await this.applicationService.createOrUpdate(application);
+    let statusId;
+    if (application.status) {
```

Comment:
do we want to check if application status changed?

Reply:
Maybe? I'm not quite sure what you are suggesting. Like throw an error if it's the same status?

Reply:
I mean like skip the status update flow if it is the same

Reply:
ah yes! Ill add that in

Reply:
Done!